aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,111 @@
+## Caching basics
 
-[!code-csharp[Main](memory/sample/src/CachingSample/Startup.cs?range=12-15&highlight=3)]
+Caching can significantly improve the performance and scalability of an app by reducing round trips to the data store. Caching works best with data that changes infrequently. Caching makes a copy of data that can be returned much faster than from the original source. You should write and test your app to never depend on cached data.
 
-You utilize caching in your app by requesting an instance of `IMemoryCache` in your controller or middleware constructor. In the sample for this article, we are using a simple middleware component to handle requests by returning customized greeting. The constructor is shown here:
+ASP.NET Core supports several different caches. The simplest cache is based on the [IMemoryCache](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache), which represents a cache stored in the memory of the web application. Apps that utilize scale out in the cloud (run on multiple servers, that is a server farm), should ensure that sessions are sticky when using the in-memory cache. Sticky sessions ensure that subsequent requests from a client all go to the same server. For example, Azure Web apps use [Application Request Routing](http://www.iis.net/learn/extensions/planning-for-arr) (ARR) to route all subsequent requests to the same server.
 
-[!code-csharp[Main](memory/sample/src/CachingSample/Middleware/GreetingMiddleware.cs?range=19-28&highlight=2,7)] 
+If an app is hosted by multiple servers in a web farm or cloud hosting environment, and you cannot guarantee sticky sessions (that is subsequent requests to the same server), you will not be able to use the in-memory cache. Non-sticky sessions require a [distributed cache](distributed.md) to avoid cache consistency problems. 
 
-## Reading and Writing to a Memory Cache
+For some apps, a [distributed cache](distributed.md) can support higher scale out than an in-memory cache. Using a distributed cache offloads the cache memory to an external process. 
 
-The middleware's `Invoke` method returns the cached data when it's available.
+The `IMemoryCache` cache will evict cache entries under memory pressure unless the [cache priority](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheitempriority) is set to `CacheItemPriority.NeverRemove`. You can set the [cache priority](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheitempriority) to adjust the priority the cache evicts items under memory pressure.
 
-There are two methods for accessing cache entries:
+The in-memory cache can store non-serializable objects, the distributed cache requires objects to be serializable.
```

Comment:
The distributed cache doesn't require objects to be serializable since we don't serialize anything explicitly. The interface is limited to `byte[]` since we expect the user to serialize their own objects.